src/Microsoft.DotNet.VersionTools/BuildManifest/Model/BuildIdentity.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.VersionTools.BuildManifest.Model
+{
+    public class BuildIdentity
+    {
+        public BuildIdentity(
+            string name,
+            string buildId,
+            string branch = null,
+            string commit = null)
+        {
+            if (name == null)
```

Comment:
Should this be IsNullOrEmpty?

Reply:
Yeah, no reason to support an empty build name. Changing.